opty/tests/test_direct_collocation.py

```diff
@@ -1907,3 +1907,405 @@ def test_attributes_read_only():
 
         with raises(AttributeError):
             setattr(test, XX, 5)
+
+def test_linear_initial_guess():
+    """Test to check if the initial guess is created correctly."""
+
+    x, y, ux, uy = mech.dynamicsymbols('x y ux uy')
+    u1, u2 = mech.dynamicsymbols('u1 u2')
+    a1, a2 = sym.symbols('a1 a2')
+    b1, b2 = sym.symbols('b1 b2')
+    t = mech.dynamicsymbols._t
+
+    # just random eoms, no physical meaning.
+    eom = sym.Matrix([
+        -x.diff(t) + a1,
+        -ux.diff(t) + u1 * b1,
+        -y.diff(t) + a1,
+        -uy.diff(t) + a2 * u2 + b2,
+    ])
+
+    state_symbols = (x, y, ux, uy)
+    par_map = {b1: 1.0,
+               b2: 2.0,
+    }
+    num_nodes = 61
+
+    # A: CONSTANT TIME INTERVAL
+    # A!: normal
+    t0, t1, t2, tf = 0.0, 2.0, 4.0, 6.0
+    interval_value = tf/(num_nodes - 1)
+
+    def obj(free):
+        Fx = free[0*num_nodes:3*num_nodes]
+        return interval_value*np.sum(Fx**2)
+
+    def obj_grad(free):
+        grad = np.zeros_like(free)
+        grad[0: 2*num_nodes] = 2.0*free[0:2*num_nodes]*interval_value
+        return grad
+
+    instance_constraints = (
+        x.func(t0) - 3.0,
+        y.func(t0) - 2.0,
+        ux.func(t0) - 5.0,
+
+        x.func(t1) - 1.5*b1,
+        y.func(t1) - 4.0,
+        ux.func(t1) - 1.0+b2,
+        uy.func(t1) - 4.5,
+
+        x.func(t2) + 2.0,
+        y.func(t2) - 3.0,
+
+        x.func(tf) + 1.0,
+        ux.func(tf) + 5.0,
+    )
+
+    bounds= {
+        a1: (-1.0, 10.0),
+        a2: (-1.0, 12.0),
+
+        u1: (-10.0, 1.0),
+        u2: (-1.0, 10.0),
+
+        x: (-5.0, 5.0 ),
+        ux: (-1.0, 1.0),
+        y: (-4.0, 4.0),
+        uy: (-1.0, 1.0),
+    }
```

Comment:
Easier to compare if placed here:

![image](https://github.com/user-attachments/assets/9d1b6b07-63e4-4f49-b591-9ce63d7b20ad)